prow/cmd/tackle/ingress_test.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	extensions "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	discoveryFake "k8s.io/client-go/discovery/fake"
+	k8sFake "k8s.io/client-go/kubernetes/fake"
+	"testing"
+	"time"
+)
+
+func TestNetworkingIngressNotExists(t *testing.T) {
+	fakeClient := k8sFake.NewSimpleClientset()
+	actual := networkingIngressAvailable(fakeClient)
+
+	if actual != false {
+		t.Fatalf("networkingIngressAvailable should return `false` if networking ingresses is unavailable")
+	}
+}
+
+func TestNetworkingIngressExists(t *testing.T) {
+	fakeClient := k8sFake.NewSimpleClientset()
+	fakeDiscovery := fakeClient.Discovery().(*discoveryFake.FakeDiscovery)
+	fakeNetworking := metav1.APIResourceList{
+		GroupVersion: "networking.k8s.io/v1beta1",
+		APIResources: []metav1.APIResource{{Name: "ingresses"}},
+	}
+	fakeDiscovery.Resources = append(fakeDiscovery.Resources, &fakeNetworking)
+
+	actual := networkingIngressAvailable(fakeClient)
+
+	if actual != true {
+		t.Fatalf("networkingIngressAvailable should return `true` if networking ingresses is available")
+	}
+}
+
+func TestIngressConversion(t *testing.T) {
```

Comment:
I do not see any useful testing happening in this method.

Reply:
I am verifying `toNewIngress` does not fail to convert and that the two different serialized lists are byte-equivalent.